src/main/proto/netflix/titus/titus_eviction_api.proto

```diff
@@ -0,0 +1,135 @@
+
+option java_multiple_files = true;
+option java_package = "com.netflix.titus.grpc.protogen";
+option java_outer_classname = "EvictionProto";
+
+option go_package = "titus";
+
+// ----------------------------------------------------------------------------
+// Eviction core data structures
+
+/// The system distruption budget provides an upper bound on allowed number of task evictions. The system
+//  disruption budget is defined at the global, tier and a capacity group level.
+message SystemDisruptionBudget {
+    /// System layer/entity refrence.
+    Reference reference = 1;
+
+    /// (Required) The admision control policy.
+    TokenBucketPolicy admissionControlPolicy = 2;
+}
+
+/// Current eviction quota at the global, tier, capacity group, and job level.
+//  For example, for the global disruption budget with the bucket size of 100 half empty, the eviction quota is 50.
+message EvictionQuota {
+    /// System layer/entity refrence.
+    Reference reference = 1;
+
+    /// (Required) Total number of tasks that can be evicted at a given point in time.
+    uint32 quota = 2;
+}
+
+// ----------------------------------------------------------------------------
+// GRPC services
+
+message CapacityGroupId {
+    string id = 1;
+}
+
+message TierId {
+    Tier tier = 1;
+}
+
+message TaskTerminateRequest {
+    string taskId = 1;
+
+    /// Textual description of why a task termination was requested.
+    string reason = 2;
+}
+
+message TaskTerminateResponse {
+    /// Set to true if the termination request was accepted.
+    bool allowed = 1;
+
+    /// If a termination request is not accepted, this field contains the reason code explaining why:
+    //  * aboveSystemDisruptionBudget
+    //  * aboveTierDisruptionBudget
+    //  * aboveCapacityGroupDisruptionBudget
+    //  * jobRelocationPolicyViolation
+    string reasonCode = 2;
+
+    /// If a termination request is not accepted, contains explanation why.
+    string reasonMessage = 3;
+}
+
+message ObserverEventRequest {
+    bool includeSnapshot = 2;
+}
+
+message EvictionServiceEvent {
+    message SnapshotEnd {
+    }
+
+    message SystemDisruptionBudgetUpdateEvent {
+        SystemDisruptionBudget current = 1;
+    }
+
+    message EvictionQuotaEvent {
+        EvictionQuota quota = 1;
+    }
+
+    message TaskTerminationEvent {
+        string taskId = 1;
+        bool approved = 2;
+    }
+
+    oneof Event {
+        SnapshotEnd snapshotEnd = 1;
+
+        SystemDisruptionBudgetUpdateEvent systemDisruptionBudgetUpdateEvent = 2;
+
+        EvictionQuotaEvent evictionQuotaEvent = 3;
+
+        TaskTerminationEvent taskTerminationEvent = 4;
+    }
+}
+
+service EvictionService {
+
+    rpc GetGlobalDisruptionBudget (google.protobuf.Empty) returns (SystemDisruptionBudget) {
+    }
+
+    rpc GetTierDisruptionBudget (TierId) returns (SystemDisruptionBudget) {
+    }
+
+    rpc GetCapacityGroupDisruptionBudget (CapacityGroupId) returns (SystemDisruptionBudget) {
```

Comment:
why not `rpc GetDistuptionBudget(Reference) returns (SystemDisruptionBudget)`?

Reply:
Good point. I will change the API.